apps/server/src/modules/oauth-provider/service/id-token.service.spec.ts

```diff
@@ -0,0 +1,303 @@
+import { LtiToolDO } from '@shared/domain/domainobject/ltitool.do';
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import { Logger } from '@src/core/logger';
+import { teamFactory } from '@shared/testing/factory/team.factory';
+import { MikroORM } from '@mikro-orm/core';
+import { setupEntities } from '@shared/testing';
+import resetAllMocks = jest.resetAllMocks;
+import clearAllMocks = jest.clearAllMocks;
+
+class IdTokenServiceSpec extends IdTokenService {
+	buildGroupsClaimSpec(teams: Team[]): GroupNameIdTuple[] {
+		return super.buildGroupsClaim(teams);
+	}
+
+	async createIframeSubjectSpec(userId: string, clientId: string): Promise<string | undefined> {
+		return super.createIframeSubject(userId, clientId);
+	}
+
+	getIframePropertiesSpec(): string {
+		return this.iFrameProperties;
+	}
+}
+
+describe('IdTokenService', () => {
+	let orm: MikroORM;
+	let module: TestingModule;
+	let idTokenService: IdTokenServiceSpec;
+	let pseudonymRepo: DeepMocked<PseudonymsRepo>;
+	let ltiToolRepo: DeepMocked<LtiToolRepo>;
+	let teamsRepo: DeepMocked<TeamsRepo>;
+	let userService: DeepMocked<UserService>;
+
+	const userId = 'userId';
+	const clientId = 'clientId';
+	const host = 'http://host';
+	let ltiToolDo: LtiToolDO;
+	let pseudonymDo: PseudonymDO;
+
+	beforeAll(async () => {
+		jest.spyOn(Configuration, 'get').mockImplementation((key: string) => {
+			switch (key) {
+				case 'HOST':
+					return host;
+				default:
+					return null;
+			}
+		});
+
+		module = await Test.createTestingModule({
+			providers: [
+				IdTokenServiceSpec,
+				{
+					provide: PseudonymsRepo,
+					useValue: createMock<PseudonymsRepo>(),
+				},
+				{
+					provide: LtiToolRepo,
+					useValue: createMock<LtiToolRepo>(),
+				},
+				{
+					provide: TeamsRepo,
+					useValue: createMock<TeamsRepo>(),
+				},
+				{
+					provide: UserService,
+					useValue: createMock<UserService>(),
+				},
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+			],
+		}).compile();
+
+		idTokenService = module.get(IdTokenServiceSpec);
+		pseudonymRepo = module.get(PseudonymsRepo);
+		ltiToolRepo = module.get(LtiToolRepo);
+		teamsRepo = module.get(TeamsRepo);
+		userService = module.get(UserService);
+		orm = await setupEntities();
+	});
+
+	afterAll(async () => {
+		await module.close();
+		await orm.close();
+		clearAllMocks();
+	});
+
+	beforeEach(() => {
+		ltiToolDo = { id: 'ltiToolId' } as LtiToolDO;
+		pseudonymDo = { pseudonym: 'Pseudonym' } as PseudonymDO;
+	});
+
+	afterEach(() => {
+		resetAllMocks();
+	});
+
+	describe('createIdToken', () => {
+		let userDto: UserDto;
+		let scopes: string[];
+
+		beforeEach(() => {
+			userDto = { id: userId, email: 'email', schoolId: 'schoolId' } as UserDto;
+			scopes = ['openid', 'offline', 'profile', 'email', 'groups'];
+			userService.getUser.mockResolvedValue(userDto);
+		});
+
+		it('should call teamsRepo if scopes contains groups', async () => {
+			// Act
```

Comment:
probably the 20th time I say this, but these comments seem very much unnecessary here

Reply:
Removed